mui/dockwidgets/state_list_widget.py

```diff
@@ -46,6 +49,30 @@ def __init__(self, name: str, parent: ViewFrame, data: BinaryView):
         self.states: Dict[int, StateDescriptor] = {}
         self.state_items: Dict[int, QTreeWidgetItem] = {}
 
+    @Slot(QTreeWidgetItem, int)
+    def onClick(self, item: QTreeWidgetItem, col: int):
+        """Jump to the current PC of a given state when double clicked"""
+
+        # do nothing on non-state items
+        if item in self.state_lists:
+            return
+
+        item_id = int(item.text(0).split(" ")[-1])
```

Comment:
This looks a little brittle. Is there any way to assign metadata for the integer state id (or maybe extend functionality of the `QTreeWidgetItem` to do this) to the state items so that we don't have to parse strings?

Reply:
I started storing the state id on `QTreeWidgetItem`s using the `setData` method. This helps avoid string parsing.